### PR TITLE
Fixed #32923 -- Refactored out Field._clean_bound_field().

### PR DESCRIPTION
**Pull request by Waheedsys**
…heck in BaseForm._clean_fields()

### CODE REVIEW

**Comment by Waheedsys** [comment]
if my guess is  correct the test failed because of my version is 5.1.dev20240117133405, but required version is 3.2  can you please give me advice on  how do i change the version of Django . [https://code.djangoproject.com/ticket/32923](url)
thank you

**Comment by felixxm** [comment]
This has nothing to do with a version on Trac. Imports are wrong in these tests, check out #17752.

**Comment by Waheedsys** [comment]
so should i commit after https://github.com/django/django/pull/17752  issue is solved

**Comment by Waheedsys** [comment]
Thanks https://github.com/felixxm

**Comment by felixxm** [comment]
@Waheedsys Can you first check Nick's [questions](https://code.djangoproject.com/ticket/32923#comment:1)? and share your thoughts.

File: django/forms/forms.py [hunk 1]
```diff
@@ -331,11 +331,7 @@ def _clean_fields(self):
             field = bf.field
             value = bf.initial if field.disabled else bf.data
             try:
-                if isinstance(field, FileField):
-                    value = field.clean(value, bf.initial)
-                else:
-                    value = field.clean(value)
-                self.cleaned_data[name] = value
+                self.cleaned_data[name] = field._clean_bound_field(bf)
```
Comment (felixxm) [inline django/forms/forms.py:334]: It looks that the following line in `_clean_fields()` is no longer needed:
```python
value = bf.initial if field.disabled else bf.data
```

**Comment by Waheedsys** [comment]
I have thoroughly reviewed Nick's questions,
in the first approach  can improve code readability by encapsulating logic within the 'bound_data' method, It may provide a cleaner and more maintainable solution, especially if the logic is complex.

in the second approach i don't have much opinion as it may impact existing code and public API.

in the third approach If passing initial into 'FileField.clean()' feels like a workaround for a specific scenario that may indicate deeper design issues

**Comment by felixxm** [comment]
@ngnpope Can you take a look?

**Comment by felixxm** [comment]
@Waheedsys Thanks :+1: 

I think it's worth doing.

**Pull request merged**